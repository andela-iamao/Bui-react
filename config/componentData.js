module.exports = [{"name":"Button","description":"Button Component","props":{"children":{"type":{"name":"node"},"required":true,"description":"Child elements for the button component"},"onClick":{"type":{"name":"func"},"required":false,"description":""},"type":{"type":{"name":"enum","value":[{"value":"'primary'","computed":false},{"value":"'secondary'","computed":false},{"value":"'danger'","computed":false}]},"required":false,"description":""}},"code":"import React from 'react';\nimport PropTypes from 'prop-types';\n\n\nconst colorScheme = {\n    primary: '#236167',\n    secondary: '#000000',\n    danger: '#AA3939'\n}\n\n/**\n * Button Component\n */\nfunction Button({ children, ...props}) {\n    return (\n        <button {...props} style={style(props.type)}>\n            {children}\n        </button>\n    );\n}\n\nconst style = (type) => ({\n    color: '#FFFFFF',\n    backgroundColor: colorScheme[type],\n    padding: 10,\n    width: 120,\n    border: 'none'\n});\n\nButton.propTypes = {\n    /**\n     * Child elements for the button component\n     */\n    children: PropTypes.node.isRequired,\n    onClick: PropTypes.func,\n    type: PropTypes.oneOf(['primary', 'secondary', 'danger'])\n}\n\nexport default Button;\n","examples":[{"name":"DangerButton","description":"Danger Button component","code":"import React from 'react';\nimport Button from 'bui-react/Button';\n\n/**\n * Danger Button component\n */\nexport default () => {\n    return (\n        <Button type=\"danger\">Danger Button!</Button>\n    );\n}\n"},{"name":"PrimaryButton","description":"Primary Button component","code":"import React from 'react';\nimport Button from 'bui-react/Button';\n\n/**\n * Primary Button component\n */\nexport default () => {\n    return (\n        <Button type=\"primary\">Primary Button!</Button>\n    );\n}\n"},{"name":"SecondaryButton","description":"Secondary Button component","code":"import React from 'react';\nimport Button from 'bui-react/Button';\n\n/**\n * Secondary Button component\n */\nexport default () => {\n    return (\n        <Button type=\"secondary\">Secondary Button!</Button>\n    );\n}\n"}]}]