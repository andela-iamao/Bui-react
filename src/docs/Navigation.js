import React from 'react';
import PropTypes from 'prop-types';

const Navigation = ({ components }) => {
    return (
        <ul className="navigation">
            {components.map(name => (
                <li key={name} className="bui-react-docs-component-name">
                    <a href={`#${name}`}>{name}</a>
                </li>
            ))}
        </ul>
    )
}

Navigation.propTypes = {
    components: PropTypes.array.isRequired
}

export default Navigation;
