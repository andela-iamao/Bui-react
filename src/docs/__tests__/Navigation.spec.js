import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';

import Navigation from '../Navigation';
import componentData from '../../../config/jest/componentDataMock';

const DATA = componentData.map((component) => component.name);


describe('DOCS: Navigation', () => {

    describe('Snapshot Test', () => {
        test('renders navigation woth a list of props', () => {
            const tree = renderer.create(
                <Navigation components={DATA} />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });


    describe('Unit Test', () => {
        test('renders a clickable link of components', () => {
            const wrapper = shallow(
                <Navigation components={DATA} />
            );
            expect(wrapper.find('li')).toHaveLength(DATA.length);
        });
    });
});