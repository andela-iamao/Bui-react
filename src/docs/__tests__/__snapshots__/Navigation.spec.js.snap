// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOCS: Navigation Snapshot Test renders navigation woth a list of props 1`] = `
<ul
  className="navigation"
>
  <li
    className="bui-react-docs-component-name"
  >
    <a
      href="#Button"
    >
      Button
    </a>
  </li>
</ul>
`;
